test/integration/MongooseAdapterSpec.js

```diff
@@ -72,4 +86,18 @@ describe('MongooseAdapterIntegration', function () {
       .catch(err => done(err))
     ;
   });
+
+  /* eslint-disable no-underscore-dangle */
+  it('allows to pass mongo ObjectId as defalt attibute', function (done) {
+    mongoUnavailable ? this.skip() : null;
+
+    factory.create('thread')
+      .then(thread => {
+        factory.create('email', { thread: thread._id }).then(email => {
+          expect(email.thread).to.be.equal(thread._id);
+          done();
+        });
+      });
+  });
```

Comment:
This way of writing specs doesn't seem to handle errors nicely. If any occurs then you just get spec timeout (unresolved promise) rather than any helpful message.
You could make use of promise chaining, so that any failure will result in calling `done(error)`.
Also promises used as above looks almost exactly like callbacks for me.
```js
let thread;
factory.create('thread')
  .then(created => thread = created)
  .then(() => factory.create('email', { thread: thread._id }))
  .then(email => expect(email.thread).to.be.equal(thread._id)
  .then(done, done);
});
```
(It's just a suggestion how I'd do this)